packages/docusaurus-theme-common/src/utils/colorModeUtils.tsx

```diff
@@ -21,67 +21,72 @@ import {createStorageSlot} from './storageUtils';
 import {useThemeConfig} from './useThemeConfig';
 
 type ColorModeContextValue = {
+  readonly colorMode: ColorMode;
+  readonly setColorMode: (colorMode: ColorMode) => void;
+
+  // TODO legacy APIs kept for retro-compatibility: deprecate them
   readonly isDarkTheme: boolean;
```

Comment:
Might just add a deprecation notice now, whenever people are trying to access `isDarkTheme` or `setLightTheme`? This can be done by returning a `get isDarkTheme() { console.warn("deprecated"); return isDarkTheme; }`

Reply:
yes 👍  I thougt it would be more complicated, maybe using a proxy or something but 🤷‍♂️ 

Going to lunch, let me know if you want to do it right now